### PR TITLE
Added MVC matching logic, tests, feature flag

### PR DESCRIPTION
**Pull request by disulliv**
- some refactoring
- fixed broken tests
- Added matching license logic
- Updated mvc description
- Added mvc tests
- Added feature flag

### CODE REVIEW

File: providers/curation/github.js [hunk 1]
```diff
@@ -414,8 +450,39 @@ class GitHubCurationService {
...
+        }
+      })
+    })
+
+    if (matchingLicenses.length > 0) {
+      output += `\nMatching license file(s): ${matchingLicenses.join(', ')}`
+    }
+
+    if (Object.keys(matchingMetadata).length > 0) {
+      const metadataText = Object.keys(matchingMetadata).length == 1
```
Comment (MichaelTsengLZ) [inline providers/curation/github.js:479]: Nit: maybe use [].join('\n')?

**Review by MichaelTsengLZ — CHANGES_REQUESTED** [review]
Overall looks good. Small things need to change.

File: providers/curation/github.js [hunk 2]
```diff
@@ -414,8 +450,39 @@ class GitHubCurationService {
     )
   }
 
-  _formatAdditionalRevisions(revisions, reason) {
-    return Object.keys(revisions).map(revision => `\n${revision}, ${revisions[revision].licensed.declared}, ${reason}`)
+  _formatMultiversionCuratedRevisions(mvcResults) {
+    let output = '**Automatically added versions:**\n'
+    mvcResults
+      .map(result => result.version)
+      .sort()
```
[the remark this answers is not in the document]
Comment (disulliv) [reply providers/curation/github.js:457]: Yep, but we can't use semver for things like Git component hashes, right? As long as there's some sort of ordering I think it's better than nothing. Lmk if you have a change in mind, or if we can keep it like this

File: providers/curation/github.js [hunk 3]
```diff
@@ -164,25 +194,24 @@ class GitHubCurationService {
...
+    const curationRevisions = get(component, 'revisions')
+    const revision = first(Object.keys(curationRevisions))
+    const componentCoordsWithRevision = { ...component.coordinates, revision }
+    const coordinates = EntityCoordinates.fromObject(componentCoordsWithRevision)
+    const revisionlessCoords = coordinates.asRevisionless()
+
+    const coordinatesList = await this.definitionService.list(revisionlessCoords)
+    const filteredCoordinatesList = coordinatesList
+      .map(stringCoords => EntityCoordinates.fromString(stringCoords))
+      .filter(coords => coordinates.name === coords.name)
```
[the remark this answers is not in the document]
Comment (disulliv) [reply providers/curation/github.js:206]: Will do

File: providers/curation/github.js (hunk 1 above)
Comment (disulliv) [reply providers/curation/github.js:479]: I was thinking for 1 metadata match it looks like
"Matching metadata: PropPath: PropValue"

But for 2 matches, it looks like
Matching metadata:
- PropPath1: PropValue1
- PropPath2: PropValue2

So if there's multiple, it has bullet points

File: providers/curation/github.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (MichaelTsengLZ) [reply providers/curation/github.js:457]: Yep, for the comparison funciton if it's not a semantic version, just return 0. If it's return `gte() ? -1 : 1`

File: providers/curation/github.js (hunk 1 above)
Comment (MichaelTsengLZ) [reply providers/curation/github.js:479]: I see. Then I think your code is fine.

**Review by MichaelTsengLZ — APPROVED** [review]

**Pull request merged**